src/sampler.jl

```diff
@@ -25,7 +25,7 @@ function sample(h::Hamiltonian, prop::AbstractProposal, θ::AbstractVector{T}, n
         if i <= n_adapts
             adapt!(adaptor, θs[i], αs[i])
             h, prop = update(h, prop, adaptor)
-            verbose && i == n_adapts && @info "Finished $n_adapts adapation steps" typeof(adaptor) prop.integrator.ϵ h.metric.sqrtM⁻¹
+            verbose && i == n_adapts && @info "Finished $n_adapts adapation steps" typeof(adaptor) prop.integrator.ϵ h.metric.M⁻¹
```

Comment:
Only `M⁻¹` is common.

Reply:
I see. Does `sqrtM⁻¹` refer to the Cholesky decomposition? If so, pls consider merging `sqrtM⁻¹` and `M⁻¹` into `PDMat`:

- `Metric.M::AbstractVector{T}` ==> `Metric.M::AbstractPDMat{T}`
- Remove `Metric.sqrtM⁻¹` since `PDMat` has the following definition:

```julia

struct PDMat{T<:Real,S<:AbstractMatrix} <: AbstractPDMat{T}
    dim::Int                    # matrix dimension
    mat::S                      # input matrix
    chol::Cholesky{T,S}         # Cholesky factorization of mat
end

```

This allow us to use existing implementation from the `PDMAts.jl` package to compute and cache `Colesky` factorization.  

See https://github.com/JuliaStats/PDMats.jl

Reply:
Cool I will take a look.

Reply:
This problem of recomputing Cholesky factorization if from the adaption interface, i.e. we try to compute a new `Hamiltonian` even the metric stays the same. Otherwise, the use of `sqrtM⁻¹` from `M⁻¹` is already cached.

Reply:
ok, I see. I meant to simplify the `Metric` type a bit by merging three different types (i.e. scalar, diagonal, and dense), through replacing our `M⁻¹::A` with `M⁻¹::AbstractPDMat`. I'll open a separate issue to explain this.